### PR TITLE
Removing incorrect comments and minor improvements

### PR DESCRIPTION
**Pull request by vinagesh**
## Description of the changes
- Removed unnecessary serialization of class properties  Did not change the class that has already been made public to keep the contract intact.
- Changed the class access modifiers explicitly to internal as that is sufficient
- Renamed the private variables in the class according to coding standards

This change was made to mainly address the TODO in one of the files that said - Encrypt the Json property. This class is internal and does not seem like it needs any seiralization. I've removed the Json attribute and removed the unnecessary TODO from the files. The code is building and E2E tests passed. If I'm missing something, please let me know and i'll make changes accordingly.

## Reference/Link to the issue solved with this PR (if any)
https://msazure.visualstudio.com/One/_workitems/edit/5867874

### CODE REVIEW

**Comment by vinagesh** [comment]
azp/run

File: iothub/service/src/Common/Security/ISharedAccessSignatureCredential.cs [hunk 1]
```diff
@@ -7,7 +7,7 @@ namespace Microsoft.Azure.Devices.Common.Security
 
     using Microsoft.Azure.Devices.Common.Data;
       
-    public interface ISharedAccessSignatureCredential
+    internal interface ISharedAccessSignatureCredential
```
Comment (abhipsaMisra) [inline iothub/service/src/Common/Security/ISharedAccessSignatureCredential.cs]: Is this change intentional?

Comment (vinagesh) [reply iothub/service/src/Common/Security/ISharedAccessSignatureCredential.cs]: yes. If we want to make SharedAccessSignatureAuthorizationRule internal, we have to make this internal. And it seems like that visibility is sufficient.

Comment (drwill-ms) [reply iothub/service/src/Common/Security/ISharedAccessSignatureCredential.cs]: Well, we shouldn't be making any types that were public not public any more, unless we have reason to believe the types were of no use.

File: iothub/service/src/Common/Data/SharedAccessSignatureAuthorizationRule.cs [hunk 2]
```diff
@@ -55,12 +48,12 @@ public bool Equals(SharedAccessSignatureAuthorizationRule other)
...
-                string.Equals(this.SecondaryKey, other.SecondaryKey, StringComparison.Ordinal) &&
-                Equals(this.Rights, other.Rights);
+            bool equals = string.Equals(KeyName, other.KeyName, StringComparison.OrdinalIgnoreCase) &&
+                string.Equals(PrimaryKey, other.PrimaryKey, StringComparison.Ordinal) &&
+                string.Equals(SecondaryKey, other.SecondaryKey, StringComparison.Ordinal) &&
+                Equals(Rights, other.Rights);
 
             return equals;
-        }
+        }
```
Comment (drwill-ms) [inline iothub/service/src/Common/Data/SharedAccessSignatureAuthorizationRule.cs]: somehow, you introduced extra space after this brace
